Use the new marathon packaging (testing with a snapshot build)

timcharper opened the pull request:
- Pkgpanda doesn't let marathon and metronome share files, so make marathon package-private
- Add some docs and fix MARATHON_CMD => MARATHON_
- Set Hostname is set via the environment

## High Level Description

## Changes from 1.4.x to 1.5.0 (unreleased)

### Breaking Changes

#### Packaging standardized

We now publish more normalized packages that attempt to follow Linux Standard Base Guidelines and use sbt-native-packager to achieve this.
As a result of this and the many historic ways of passing options into marathon, we will only read `/etc/default/marathon` when starting up.
This file, like `/etc/sysconfig/marathon`, has all marathon command line options as "MARATHON_XXX=YYY" which will translate to `--xx=yyy`.
We no longer support /etc/marathon/conf which was a set of files that would get translated into command line arguments. In addition,
we no longer assume that if there is no zk/master argument passed in, then both are running on localhost.

If support for any of the above is important to you, please file a JIRA and/or create a PR/Patch.


#### App JSON Fields Changed or Moved.

Marathon will continue to *accept* the app JSON as it did in 1.4;
however, applications that use deprecated fields will be normalized into a canonical representation.
The app JSON *generated* by the /v2 REST API has changed: only canonical fields are generated.
The [App RAML specification](docs/docs/rest-api/public/api/v2/types/app.raml) is the source of truth with respect to deprecated fields.
The following deprecated fields will no longer be generated for app JSON:

- `ipAddress`
- `container.docker.portMappings`
- `container.docker.network`
- `ports`
- `uris`

Marathon clients that consume these deprecated fields will require changes.
In addition, new networking API fields have been introduced:

- `networks`
- `container.portMappings`

The `networks` field replaces the `ipAddress.networkName` and `container.docker.network` fields, and supports joining an app to multiple `container` networks.
The legacy IP/CT API did not require a resolvable network name in order to use a `container` network;
it allowed both an app definition to leave `ipAddress.networkName` unspecified **and** the operator to leave `--default_network_name` unspecified.
Starting with Marathon v1.5 such apps will be rejected: apps may leave `networks[x].name` unspecified for `container` networks only if `--default_network_name` has been specified by the operator.
Marathon injects the value of `--default_network_name` into unnamed `container` networks upon app create/update.

Upgrading from Marathon 1.4.x to Marathon 1.5.x will automatically migrate existing applications to the new networking API.
Migration of legacy Mesos IP/CT apps **may fail** if those apps did not specify `ipAddress.networkName` and there is no default network name specified.
See the (networking documentation)[docs/docs/networking.md] for details concerning app migration and network API changes.

The [old app networking docs](docs/docs/ports.md) have been relocated.
See the [networking documentation](docs/docs/networking.md) for details concerning the new API.

#### Metric Names Changed or Moved.
We moved to a different Metrics library and the metrics are not _always_ compatible or the same as existing metrics;
however, the metrics are also now more accurate, use less memory, and are expected to get better throughout the release.
Where it was possible, we maintained the original metric names/groupings/etc, but some are in new locations or have
slightly different semantics. Any monitoring dashboards should be updated.

Before 1.5.0 releases, we will publish a migration guide for the new metric formats and where the replacement
metrics can be found and the formats they are now in.

#### Artifact store has been removed
The artifact store was deprecated with Marathon 1.4 and is removed in version.
The command line flag `--artifact_store` will throw an error if specified.
The REST API endpoint `/v2/artifacts` has been removed completely.

#### Logging endpoint
Marathon has the ability to view and change log level configuration during runtime via the `/logging` endpoint.
This version switches from a form based API to a JSON based API, while maintaining the functionality.
We also secured this endpoint, so you can restrict who is allowed to view or update this configuration.
Please find our [API documentation](https://mesosphere.github.io/marathon/api-console/index.html) for all details.

#### Event Subscribers has been removed.
The events subscribers endpoint (`/v2/eventSubscribers`) was deprecated in Marathon 1.4 and is removed in this version.
Please move to the `/v2/events` endpoint instead.

#### Removed command line parameters
- The command line flag `max_tasks_per_offer` has been deprecated since 1.4 and is removed now. Please use `max_instances_per_offer`.

#### Deprecated command line parameters
- The command line flag `save_tasks_to_launch_timeout` is deprecated and has no effect any longer.

### New Features

#### Networking Improvements Involving Multiple Container Networks

The field `networkNames` has been added to [app container's ContainerPortMapping](docs/docs/rest-api/public/api/v2/types/appContainer.raml) and [pod's Endpoint](docs/docs/rest-api/public/api/v2/types/network.raml). Using the field, an app or pod participating in multiple container networks can now forward ports by specifying a single item `networkNames`. For more information, see the [networking documentation](./docs/docs/networking.md).

Additionally container port discovery has been improved, with a pod or app being able specify with which container network(s) a port name/protocol/etc is associated. Discovery labels are now generated for container networks associated with ports.

#### Mesos Bridge Network Name Configurable

The CNI network used for Mesos containers when bridge networking is now configurable via the command-line argument `--mesos_bridge_name`. As with other command-line-args, this can also be specified via `MARATHON_MESOS_BRIDGE_NAME`, as well.

#### Backup and Restore Operations

You can now backup and restore Marathon's internal state via the [DELETE /v2/leader](./docs/docs/rest-api/public/api/v2/leader.raml) API endpoint.

See [MARATHON-7041](https://jira.mesosphere.com/browse/MARATHON-7041)

#### TTY support

You can now specify that a TTY should be allocated for app or pod containers. See the [TTY definition](./docs/docs/rest-api/public/api/v2/types/containerCommons.raml). An example can be found in [v2/examples/app.json](./docs/docs/rest-api/public/api/v2/examples/app.json).

See [MARATHON-7062](https://jira.mesosphere.com/browse/MARATHON-7062)

#### Improved Validation Error Messages

All validation specified in the RAML is now programatically enforced, leading to more consistent, descriptive, and legible error messages.

#### Security improvements

Marathon is in better compliance with various security best-practices. An example of this is that Marathon no longer responds to the directory listing request.

### Fixed issues
- [MARATHON-7320](https://jira.mesosphere.com/browse/MARATHON-7320) Fix MAX_PER constraint for attributes.

------------------------------------------------------------

## Related Issues

- See high level description

## Checklist for all PR's

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review): [commits](https://github.com/mesosphere/marathon/compare/releases/1.4...master)
  - [ ] Test Results: [Master Marathon CI](https://jenkins.mesosphere.com/service/jenkins/view/Marathon/job/marathon-pipelines/job/master/25/)
  - [ ] Code Coverage (if available): ¯\_(ツ)_/¯ we track this but I don't know where it is to link. It's good. Trust me.

klueska commented:
With this newest PR I get marathon failing with:
```
Apr 27 23:51:59 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal systemd[1]: Stopped Marathon: container orchestration engine.
Apr 27 23:51:59 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal systemd[1]: Starting Marathon: container orchestration engine...
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [INFO] Clearing proxy environment variables
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [INFO] PID 4558 has command line [b'/opt/mesosphere/active/java/usr/java/bin/java', b'-Dzookeeper.log.dir=/var/lib/dcos/exhibitor/zookeeper', b'-Dzookeeper.root.logger=INFO,CONSOLE', b'-cp', b'/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../build/classes:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../build/lib/*.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/slf4j-log4j12-1.6.1.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/slf4j-api-1.6.1.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/netty-3.7.0.Final.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/log4j-systemd-journal-appender-1.3.2.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/log4j-jna-4.2.2.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/log4j-1.2.16.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../lib/jline-0.9.94.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../zookeeper-3.4.8.jar:/opt/mesosphere/active/exhibitor/usr/zookeeper/bin/../src/java/lib/*.jar:/var/lib/dcos/exhibitor/conf:', b'-Djna.tmpdir=/var/lib/dcos/exhibitor/tmp', b'-Dcom.sun.management.jmxremote', b'-Dcom.sun.management.jmxremote.local.only=false', b'org.apache.zookeeper.server.quorum.QuorumPeerMain', b'/var/lib/dcos/exhibitor/conf/zoo.cfg']
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [INFO] PID file hasn't been modified. ZK still seems to be at that PID.
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [INFO] Shortcut succeeeded, assuming local zk is in good config state, not waiting for quorum.
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [INFO] Connecting to 127.0.0.1:2181
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [INFO] Zookeeper connection established, state: CONNECTED
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal bootstrap[12262]: [DEBUG] bootstrapping dcos-marathon
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal systemd[1]: Started Marathon: container orchestration engine.
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal systemd[1]: dcos-marathon.service: Main process exited, code=exited, status=1/FAILURE
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal systemd[1]: dcos-marathon.service: Unit entered failed state.
Apr 27 23:52:00 klueska-e50e2860-masters-zgpl.c.massive-bliss-781.internal systemd[1]: dcos-marathon.service: Failed with result 'exit-code'.
```

orlandohohmeier commented:
Marathon 1.5+ breaks the UI due to the networking related API changes thus shouldn't be merged until the respective UI issues have been addressed.

asridharan commented:
@timcharper looks like integration tests are failing on a bunch of different suites, is this ready to merge, or should we updating the snapshot on Marathon. Wanted to figure out which snapshot to use for testing UCR bridge mode.

kensipe commented:
@asridharan we are not ready to merge... we are triaging these failed tests.

timcharper commented:
There was discussion that the purpose of the build is to unblock the UI team (and others?), and not to create a release for others to start playing with.

I don't know what that means regarding whether it should be merged or not, but hopefully it helps.

kensipe commented:
https://jira.mesosphere.com/browse/MARATHON-7318
needs to be resolved for the marathon 1.5 release on this PR

unterstein commented on gen/dcos-config.yaml:
```diff
@@ -847,7 +847,7 @@ package:
 {% switch dcos_overlay_enable %}
 {% case "true" %}
-      MARATHON_CMD_DEFAULT_NETWORK_NAME={{ dcos_overlay_network_default_name }}
+      MARATHON_DEFAULT_NETWORK_NAME={{ dcos_overlay_network_default_name }}
```
awesome!

timcharper commented:
Here's one of the failures:

```
>               raise AssertionError('Application deployment failed, reason: {}'.format(message))
E               AssertionError: Application deployment failed, reason: Failed to create docker run options: Port mappings are only supported for bridge and user-defined networks
```

🤔

timcharper commented:
@orlandohohmeier ^ it looks much better now

Failure looks like it is maybe cosmos related

```
test_composition.py:213: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def check_signal_data():
        # Check the entire hash of diagnostics data
        assert r_data['diagnostics'] == exp_data['diagnostics']
        # Check a subset of things regarding Mesos that we can logically check for
        framework_names = [x['name'] for x in r_data['mesos']['properties']['frameworks']]
        assert 'marathon' in framework_names
        assert 'metronome' in framework_names
        # There are no packages installed by default on the integration test, ensure the key exists
>       assert len(r_data['cosmos']['properties']['package_list']) == 0
E       TypeError: 'NoneType' object is not subscriptable
```

timcharper commented:
`integration-test/upgrade-onprem` continues to fail. I logged in to the test cluster and ran this command to see why:

```
[root@ip-10-10-0-157 ~]# /opt/mesosphere/bin/dcos-signal -test
ERRO[0000] Error setting track for cosmos: Response HTTP/1.1 http://127.0.0.1:7070/package/list: 400 Bad Request
ERRO[0000] Error setting track for cosmos: cosmos report is nil, bailing out.
ERRO[0000] cosmos: Response HTTP/1.1 http://127.0.0.1:7070/package/list: 400 Bad Request
ERRO[0000] cosmos: cosmos report is nil, bailing out.
{
    "cosmos": null,
    "diagnostics": {
        "properties": {
            "clusterId": "efb6b3eb-16d9-4115-9816-222841face88",
            "customerKey": "",
            "environmentVersion": "",
            "health-unit-dcos-3dt-service-total": 6,
            "health-unit-dcos-3dt-service-unhealthy": 0,
            "health-unit-dcos-3dt-socket-total": 6,
            "health-unit-dcos-3dt-socket-unhealthy": 0,
            "health-unit-dcos-adminrouter-agent-reload-service-total": 3,
            "health-unit-dcos-adminrouter-agent-reload-service-unhealthy": 0,
            "health-unit-dcos-adminrouter-agent-reload-timer-total": 3,
            "health-unit-dcos-adminrouter-agent-reload-timer-unhealthy": 0,
            "health-unit-dcos-adminrouter-agent-service-total": 3,
            "health-unit-dcos-adminrouter-agent-service-unhealthy": 0,
            "health-unit-dcos-adminrouter-reload-service-total": 3,
            "health-unit-dcos-adminrouter-reload-service-unhealthy": 0,
            "health-unit-dcos-adminrouter-reload-timer-total": 3,
            "health-unit-dcos-adminrouter-reload-timer-unhealthy": 0,
            "health-unit-dcos-adminrouter-service-total": 3,
            "health-unit-dcos-adminrouter-service-unhealthy": 0,
            "health-unit-dcos-cosmos-service-total": 3,
            "health-unit-dcos-cosmos-service-unhealthy": 0,
            "health-unit-dcos-docker-gc-service-total": 3,
            "health-unit-dcos-docker-gc-service-unhealthy": 0,
            "health-unit-dcos-docker-gc-timer-total": 3,
            "health-unit-dcos-docker-gc-timer-unhealthy": 0,
            "health-unit-dcos-epmd-service-total": 6,
            "health-unit-dcos-epmd-service-unhealthy": 0,
            "health-unit-dcos-exhibitor-service-total": 3,
            "health-unit-dcos-exhibitor-service-unhealthy": 0,
            "health-unit-dcos-gen-resolvconf-service-total": 6,
            "health-unit-dcos-gen-resolvconf-service-unhealthy": 0,
            "health-unit-dcos-gen-resolvconf-timer-total": 6,
            "health-unit-dcos-gen-resolvconf-timer-unhealthy": 0,
            "health-unit-dcos-history-service-total": 3,
            "health-unit-dcos-history-service-unhealthy": 0,
            "health-unit-dcos-log-agent-service-total": 3,
            "health-unit-dcos-log-agent-service-unhealthy": 0,
            "health-unit-dcos-log-agent-socket-total": 3,
            "health-unit-dcos-log-agent-socket-unhealthy": 0,
            "health-unit-dcos-log-master-service-total": 3,
            "health-unit-dcos-log-master-service-unhealthy": 0,
            "health-unit-dcos-log-master-socket-total": 3,
            "health-unit-dcos-log-master-socket-unhealthy": 0,
            "health-unit-dcos-logrotate-agent-service-total": 3,
            "health-unit-dcos-logrotate-agent-service-unhealthy": 0,
            "health-unit-dcos-logrotate-agent-timer-total": 3,
            "health-unit-dcos-logrotate-agent-timer-unhealthy": 0,
            "health-unit-dcos-logrotate-master-service-total": 3,
            "health-unit-dcos-logrotate-master-service-unhealthy": 0,
            "health-unit-dcos-logrotate-master-timer-total": 3,
            "health-unit-dcos-logrotate-master-timer-unhealthy": 0,
            "health-unit-dcos-marathon-service-total": 3,
            "health-unit-dcos-marathon-service-unhealthy": 0,
            "health-unit-dcos-mesos-dns-service-total": 3,
            "health-unit-dcos-mesos-dns-service-unhealthy": 0,
            "health-unit-dcos-mesos-master-service-total": 3,
            "health-unit-dcos-mesos-master-service-unhealthy": 0,
            "health-unit-dcos-mesos-slave-public-service-total": 1,
            "health-unit-dcos-mesos-slave-public-service-unhealthy": 0,
            "health-unit-dcos-mesos-slave-service-total": 2,
            "health-unit-dcos-mesos-slave-service-unhealthy": 0,
            "health-unit-dcos-metrics-agent-service-total": 3,
            "health-unit-dcos-metrics-agent-service-unhealthy": 0,
            "health-unit-dcos-metrics-agent-socket-total": 3,
            "health-unit-dcos-metrics-agent-socket-unhealthy": 0,
            "health-unit-dcos-metrics-master-service-total": 3,
            "health-unit-dcos-metrics-master-service-unhealthy": 0,
            "health-unit-dcos-metrics-master-socket-total": 3,
            "health-unit-dcos-metrics-master-socket-unhealthy": 0,
            "health-unit-dcos-metronome-service-total": 3,
            "health-unit-dcos-metronome-service-unhealthy": 0,
            "health-unit-dcos-navstar-service-total": 6,
            "health-unit-dcos-navstar-service-unhealthy": 0,
            "health-unit-dcos-oauth-service-total": 3,
            "health-unit-dcos-oauth-service-unhealthy": 0,
            "health-unit-dcos-pkgpanda-api-service-total": 6,
            "health-unit-dcos-pkgpanda-api-service-unhealthy": 0,
            "health-unit-dcos-rexray-service-total": 3,
            "health-unit-dcos-rexray-service-unhealthy": 0,
            "health-unit-dcos-signal-service-total": 3,
            "health-unit-dcos-signal-service-unhealthy": 0,
            "health-unit-dcos-signal-timer-total": 6,
            "health-unit-dcos-signal-timer-unhealthy": 0,
            "health-unit-dcos-spartan-service-total": 6,
            "health-unit-dcos-spartan-service-unhealthy": 0,
            "health-unit-dcos-spartan-watchdog-service-total": 6,
            "health-unit-dcos-spartan-watchdog-service-unhealthy": 0,
            "health-unit-dcos-spartan-watchdog-timer-total": 6,
            "health-unit-dcos-spartan-watchdog-timer-unhealthy": 0,
            "platform": "aws",
            "provider": "onprem",
            "source": "cluster",
            "variant": "open"
        },
        "anonymousId": "efb6b3eb-16d9-4115-9816-222841face88",
        "event": "health"
    },
    "mesos": {
        "properties": {
            "agents_active": 3,
            "agents_connected": 3,
            "clusterId": "efb6b3eb-16d9-4115-9816-222841face88",
            "cpu_total": 12,
            "cpu_used": 4.4,
            "customerKey": "",
            "disk_total": 306252,
            "disk_used": 0,
            "environmentVersion": "",
            "framework_count": 2,
            "frameworks": [
                {
                    "name": "metronome"
                },
                {
                    "name": "marathon"
                }
            ],
            "mem_total": 43815,
            "mem_used": 640,
            "platform": "aws",
            "provider": "onprem",
            "source": "cluster",
            "task_count": 36,
            "variant": "open"
        },
        "anonymousId": "efb6b3eb-16d9-4115-9816-222841face88",
        "event": "mesos_track"
    }
}

[root@ip-10-10-0-157 ~]#
```

I am very convinced this is not a Marathon related failure. What is the best team / person to investigate this test failure?

timcharper commented:
What is strange is that I deployed a fresh dcos cluster using this build (not from the tests), and the above `dcos-signal -test` command succeeded, and returned a cosmos data structure as expected. I don't know what about the integration test is causing this to fall over.

timcharper commented:
Okay, I have collaborated with José in the Cosmos team; known issue.

Once https://github.com/dcos/dcos/pull/1549 is merged, then we need to rebase this PR on top of it.

timcharper commented:
I've rebased the change on top of the latest master and squashed everything down to a single commit. This should pass now that the Cosmos change #1549 is merged.

timcharper commented:
Failing because of the pyinstaller change.

From #1571 I read that we can cherry-pick https://github.com/dcos/dcos/commit/773227c7e93329e810f28ad5c6927b8670f612c4 temporarily and become unblocked.

timcharper commented:
Not sure why bumping pyinstaller caused this many failures; this was passing just a few days ago

timcharper commented:
I am told that these tests are flaky and need to be run a few times

timcharper commented:
Rebased against latest master in order to get changes from #1557, which I'm told could help with the failures.

timcharper commented:
These tests appear to be performing their function; something is broken, and, that breakage appears to be introduced with this change. Further investigation required. Paused for time being due to other higher priorities.

timcharper commented:
There is a legitimate regression as a result of the networking API changes. Network labels are being dropped during normalization. See https://jira.mesosphere.com/browse/MARATHON-7421 for more details and status of the fix.

orlandohohmeier commented:
I've pushed an update to switch DC/OS UI to the Marathon 1.5 integration branch. We should merge dcos/dcos-ui#2200 and update the DC/OS UI buildinfo before we land this PR.

orsenthil commented:
@timcharper  - why is the conflict in this?

```
Conflicting files
packages/dcos-ui/buildinfo.json
```

adam-mesos commented on packages/marathon/build:
```diff
@@ -24,20 +35,21 @@ EnvironmentFile=/opt/mesosphere/environment
-    --hostname "\$MARATHON_HOSTNAME" \\
     --default_accepted_resource_roles "*" \\
     --mesos_role "slave_public" \\
-    --max_tasks_per_offer 100 \\
```
What happened to `max_tasks_per_offer` and `hostname`? Are those now the defaults in Marathon 1.5?

timcharper replied:
Oh, geez; I assumed it was removed. But `max_instances_per_offer` has replaced it.

timcharper replied:
I'm not sure why `--hostname` was removed from the script; I didn't originally author that aspect of the change... I took it over from a former team member. I'll try and find out.

timcharper replied:
`--hostname` was removed because `$MARATHON_HOSTNAME` is implicitly passed as that parameter, and it was redundant.

`--max_instances_per_offer` has been fixed

adam-mesos commented on packages/dcos-ui/buildinfo.json:
```diff
@@ -3,7 +3,7 @@
-    "ref": "89b9fbd36582d107e67bea114151c4c5504b93a3",
-    "ref_origin": "release/1.9"
+    "ref": "62d63a79793e428596c7c631b12db2d7d68ab170",
+    "ref_origin": "integrate/marathon-1.5"
```
Please don't bump the UI here unless you need to. Remove this change if you can.
And if you do, then you should bump to a release branch and also bump it in EE.

orlandohohmeier replied:
Please note that the UI is not compatible with Marathon 1.5 without the changes introduced `integrate/marathon-1.5`, I'm ok with landing those changes in separate PRs if that makes it easier to land this PR.

timcharper replied:
Thanks @orlandohohmeier ; Anything that will help this PR get landed faster I think will make everything easier.

adam-mesos commented:
Also, since you are bumping 3 components (Marathon, Marathon plugins, and dcos-ui), please fill in the Component checklist section for each bump! How else am I supposed to know whether you changed 1 line in dcos-ui or thousands?

timcharper commented:
@adam-mesos per our discussion I have removed the UI bump to help this land; I've responded to your comments and have rebased against the latest.

Getting the tests to pass has been consistently no small effort. If you could provide a review before we expend that effort, this would be helpful :)

I will bump the EE change once this is approved and the tests pass. It seems like wasted effort to do it otherwise.

orsenthil commented:
@mesosphere-mergebot label ship-it.

adam-mesos commented:
Still need to fix up the EE PR https://github.com/mesosphere/dcos-enterprise/pull/857 before we can add them both to the next pair of trains. "Holding" until then

orsenthil commented:
@mesosphere-mergebot label holding.

jgehrcke commented:
Now that this landed -- which JIRA tickets neet to be updated (I get the feeling not all relevant tickets are linked from this PR)?
